packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -91,139 +117,535 @@ const propTypes = {
    * The selected or entered date value to display in the date input.
    */
   value: PropTypes.string,
-  /**
-  * String that labels the current element. 'aria-label' must be present,
-  * for accessibility.
-  */
-  ariaLabel: PropTypes.string,
-  /**
-   * Callback ref to pass into the input dom element.
-   */
-  inputRefCallback: PropTypes.func,
 };
 
 const defaultProps = {
+  ariaLabel: undefined,
   buttonRefCallback: undefined,
+  id: undefined,
   inputAttributes: undefined,
+  inputRefCallback: undefined,
   isIncomplete: false,
   isInvalid: false,
   name: undefined,
   onBlur: undefined,
+  onButtonFocus: undefined,
   onChange: undefined,
   onClick: undefined,
   onFocus: undefined,
-  onButtonFocus: undefined,
   onKeyDown: undefined,
   required: false,
   useExternalFormatMask: false,
   value: undefined,
-  ariaLabel: undefined,
 };
 
-class DatePickerInput extends React.Component {
-  constructor(props) {
-    super(props);
+const dateReducer = (state, inputValue) => ({
+  ...state,
+  ...inputValue,
+});
```

Comment:
I'm on the fence about this reducer implementation. I think this does work for the use-cases of this component, but this isn't really how a reducer is intended to be used. There are no dispatch actions here; all actions are value merges. The state keys and values are non-deterministic from the perspective of the reducer.